models/BaseEntity.cfc

```diff
@@ -2380,14 +2455,51 @@ component accessors="true" {
 		if ( variables._meta.originalMetadata.keyExists( "grammar" ) ) {
 			variables._builder.setGrammar( variables._wirebox.getInstance( variables._meta.originalMetadata.grammar ) );
 		}
-		retrieveQuery().from( tableName() );
-		return variables._builder
+
+		variables._builder
 			.setEntity( this )
 			.setReturnFormat( "array" )
 			.setDefaultOptions( variables._queryOptions )
-			.from( tableName() )
-			.newQuery()
-			.select( retrieveQualifiedColumns() );
+			.from( tableName() );
+
+		if ( variables._hasParentEntity ) {
+			return variables._builder
+				.newQuery()
+				.select( retrieveQualifiedColumns() )
+				.join(
+					variables._parentEntity.meta.table,
+					variables._parentEntity.meta.table & "." & variables._parentEntity.key,
```

Comment:
Again, I would love to use the parent entity's `qualifyColumn` method here somehow.  Not essential, but it would be nice.

Reply:
See note above.  `variables._parentEntity` is a struct containing the join/discriminator info and the parent meta